portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +25,32 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    //data structure to hold all comments:
+    private Vector allComments = new Vector();
+
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    //get the input from the form
+    String comment = getParameter(request, "user-comment", "");
+
+    if(comment != "") {
+        allComments.add(comment);
+    }
+
     response.setContentType("text/html;");
-    response.getWriter().println("Hello! -Hadley");
+    response.sendRedirect("/index.html#comments");    
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
```

Comment:
nit: if you want to do this in one line:
return value == null ? defaultValue : value;

Reply:
Using the if statement makes it more readable for me, so I would like to keep as is. If you think that it is more effective the other way, I can change it too.